views/natural_attractions_views.py

```diff
@@ -74,37 +75,35 @@ def get_natural_attraction_by_park_id(park_id):
     return park_natural_attractions
 
 def create_natural_attraction(new_attraction):
-    with sqlite3.connect("./national_park.sqlite3") as conn:
-        db_cursor = conn.cursor()
+    """Function to create a new resource in the database"""
 
-        db_cursor.execute("""
+    sql = """
         INSERT INTO Natural_Attractions
             ( name)
         VALUES
             ( ?);
-        """, (new_attraction['name'],))
+        """
+    sql_values =(new_attraction['name'],)
 
-        id = db_cursor.lastrowid
-        new_attraction['id'] = id
+    new_resource = create_resource(sql, sql_values, new_attraction)
 
-    return new_attraction
+    return new_resource
 
 
 def update_natural_attraction(id, new_attraction):
-    """iterates the list of wildlife until it finds the right one, and then replaces it with what the client sent as the replacement."""
-    with sqlite3.connect("./national_park.sqlite3") as conn:
-        db_cursor = conn.cursor()
+    """Function to update the Natural_Attractions table in a database"""
 
-        db_cursor.execute("""
+    sql = """
         UPDATE Natural_Attractions
             SET
                 name = ?
         WHERE id = ?
-        """, (new_attraction['name'], id, ))
+        """ 
+    sql_values = (new_attraction['name'], id, )
 
         # Were any rows affected?
         # Did the client send an `id` that exists?
-        rows_affected = db_cursor.rowcount
+    rows_affected = update_resource(sql, sql_values)
 
     if rows_affected == 0:
```

Comment:
Recommend moving 108-112 into the update_resource helper function since it can be applied generically to all modules.